### PR TITLE
OBSDATA-4029 Prevent multiple checkpoints

### PR DESCRIPTION
**Pull request by saithal-confluent**
- Prevent already checkpointed sequence from being checkpointed again even though the time has elapsed. 

#### Release note

For tips about how to write a good release note, see [Release notes](https://github.com/apache/druid/blob/master/CONTRIBUTING.md#release-notes).

<hr>

##### Key changed/added classes in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java [hunk 1]
```diff
@@ -729,7 +729,8 @@ public void onFailure(Throwable t)
           }
 
           if (System.currentTimeMillis() > nextCheckpointTime) {
-            sequenceToCheckpoint = getLastSequenceMetadata();
+            sequenceToCheckpoint = getLastSequenceMetadata().isCheckpointed() ? null : getLastSequenceMetadata();
```
Comment (m-ghazanfar) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java:732]:
```suggestion
            var lastSequenceMetadata = getLastSequenceMetadata();
            sequenceToCheckpoint = lastSequenceMetadata.isCheckpointed() ? null : lastSequenceMetadata;
```

**Review by m-ghazanfar — DISMISSED** [review]

File: indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java [hunk 2]
```diff
@@ -731,7 +731,8 @@ public void onFailure(Throwable t)
           }
 
           if (System.currentTimeMillis() > nextCheckpointTime) {
-            sequenceToCheckpoint = getLastSequenceMetadata();
+            var lastSequenceMetadata = getLastSequenceMetadata();
```
Comment (pagrawal10) [inline indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java:734]: Change to 
`SequenceMetadata<PartitionIdType, SequenceOffsetType> lastSequenceMetadata = getLastSequenceMetadata();`

Comment (m-ghazanfar) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java]: See Herb Sutter's arguments on `auto` the same apply here.

Comment (m-ghazanfar) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java]: See Herb Sutter's arguments on `auto` in this video: https://youtu.be/xnqTKD8uD64?t=1703
While he talks about this in context of C++, the same arguments apply here.

Comment (pagrawal10) [reply indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java]: The semaphore build was failing due to "var"

**Pull request closed without merging**